### PR TITLE
keep PR alive

### PR DESCRIPTION
**Pull request by ericcf**

### CODE REVIEW

File: app/scripts/controllers/usersetup.js [hunk 1]
```diff
@@ -26,6 +27,37 @@
         $scope.formInfo = {};
       }
 
+      if ($scope.purpleRobotIsAlive == null) {
```
Comment (michaelwehrley) [inline app/scripts/controllers/usersetup.js]: Did you use `==` because `null` is a primitive?

File: app/views/usersetup.html [hunk 2]
```diff
@@ -1,96 +1,102 @@
...
 
-  <div class="container">
-    <div class="form-group col-xs-8">
-      <label for="nickname" class="control-label">Nickname</label>
-      <input class="form-control" ng-required="true" id="nickname" placeholder="Nickname" ng-model="formInfo.nickname" required>
+  <form role="form" name="forms.userForm" ng-Controller="UsersetupCtrl">
+    <div class="container">
+      <div class="form-group col-xs-8">
+        <label for="configurationToken" class="control-label">Configuration Token</label>
+        <input type="text" step="1.0" class="form-control" id="configurationToken" placeholder="Configuration Token" ng-model="formInfo.configurationToken">
```
Comment (michaelwehrley) [inline app/views/usersetup.html]: Do you want `type="number"` here?

File: plugins/edu.northwestern.cbits.PurpleRobotIntent/src/android/PurpleRobotIntent.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    "edu.northwestern.cbits.purple_robot_manager";
+
+  public static final Uri PR_LAST_KNOWN_PROBE_VALUES = Uri.parse("content://edu.northwestern.cbits.purple_robot_manager.content/recent_probe_values");
+
+  @Override
+  public boolean execute(String action, String _rawArgs,
+                         CallbackContext _callbackContext) {
+    boolean result = false;
+
+    if (SEND.equals(action)) {
```
Comment (michaelwehrley) [inline plugins/edu.northwestern.cbits.PurpleRobotIntent/src/android/PurpleRobotIntent.java]: It is funny to see the Yoda conditions.  I think you expressed the need for this though.

File: app/scripts/controllers/usersetup.js (hunk 1 above)
Comment (ericcf) [inline app/scripts/controllers/usersetup.js]: In JavaScript, this is a shortcut for `x === null || x === undefined`.

Comment (michaelwehrley) [inline app/scripts/controllers/usersetup.js]: :sunglasses:

File: app/views/usersetup.html (hunk 2 above)
Comment (ericcf) [inline app/views/usersetup.html]: No, the configuration token can have letters and symbols as well.

**Comment by michaelwehrley** [comment]
It looks like all the js you added were private methods and thus no specs truly needed. :+1:

File: plugins/edu.northwestern.cbits.PurpleRobotIntent/src/android/PurpleRobotIntent.java (hunk 3 above)
Comment (ericcf) [inline plugins/edu.northwestern.cbits.PurpleRobotIntent/src/android/PurpleRobotIntent.java]: Yes, in Java, a `String` variable could be `Null` which would throw an exception if you called `.equals()` on it.

File: app/views/usersetup.html (hunk 2 above)
Comment (michaelwehrley) [inline app/views/usersetup.html]: So why `step="1.0"`?

**Pull request merged**